Collector/Collector.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Http\HttplugBundle\Collector;
+
+use Exception;
+use Http\Client\Common\Plugin;
+use Psr\Http\Message\RequestInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+
+class Collector extends DataCollector
+{
+    /**
+     * @var Profile[]
+     */
+    private $profiles;
+
+    public function __construct()
+    {
+        $this->profiles = [];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function collect(Request $request, Response $response, Exception $exception = null)
+    {
+        // We do not need to collect any data from the Symfony Request and Response
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'httplug';
+    }
+
+    /**
+     * @param Plugin           $plugin
+     * @param RequestInterface $request
+     *
+     * @return Profile
+     */
+    public function getProfile(Plugin $plugin, RequestInterface $request)
+    {
+        $key = spl_object_hash($plugin).'-'.spl_object_hash($request);
```

Comment:
A profile is the result of one plugin execution. However this lack storage is not perfect as executing the same request will override values. I probably should not hash profile to avoid such case.